lesson01.calculator/index.html

```diff
@@ -1,22 +1,210 @@
+                </div>
+            </form>
+        </div>
     </main>
+    <script type="text/javascript">
+        'use strict'
+        const CalculatorState = function () {
+            this.clear = function () {
+                this.right_value = 0;
+                this.left_value = 0;
+                this.operator = '';
+                this.resetDecimal();
+            };
+            this.resetDecimal = function () {
+                this.decimal = false;
+                this.decimal_place = 10;
+            };
+            this.clear();
+        };
+
+        let calc_state = new CalculatorState();
+
+        function putValue(value) {
+            let side = (calc_state.operator === '') ? 'left_value' : 'right_value';
+
+            if (/\d/.test(value)) {
+                // handle operand
+                value = new Number(value);
+                if (calc_state.decimal === false) {
+                    calc_state[side] *= 10;
+                    calc_state[side] += value;
+                } else {
+                    value /= calc_state.decimal_place;
+                    calc_state.decimal_place *= 10;
+                    calc_state[side] += value;
+                }
+            } else if (value === '.') {
+                calc_state.decimal = true;
+            } else {
+                // handle operator, calculate previous
+                if (calc_state.operator !== '') {
+                    try {
+                        calculate();
+                        side = 'left_value';
+                        if (value !== '=') {
+                            calc_state.operator = value;
+                        }
+                    } catch (err) {
+                        alert(err.message);
+                        clearInput();
+                    }
+                } else if (value !== '=') {
+                    calc_state.operator = value;
+                    calc_state.resetDecimal();
+                }
+            }
+            calculator.inputarea.value = calc_state[side];
+        };
+
+        function calculate() {
+            let result = 0;
+            switch (calc_state.operator) {
+                case '+':
+                    result = add(calc_state.left_value, calc_state.right_value);
+                    break;
+                case '-':
+                    result = subtract(calc_state.left_value, calc_state.right_value);
+                    break;
+                case '*':
+                    result = multiply(calc_state.left_value, calc_state.right_value);
+                    break;
+                case '/':
+                    result = divide(calc_state.left_value, calc_state.right_value);
+                    break;
+            }
+            calc_state.clear();
+            calc_state.left_value = result;
+        }
+
+        function clearInput() {
+            calc_state.clear();
+            calculator.inputarea.value = "";
+        };
+
+        function add(a, b) {
```

Comment:
all these function (apart from divide) are very short and can be included in `calculate` function.